netsim/modules/stp.py

```diff
@@ -17,6 +17,13 @@ def supports_stp(intf: Box) -> bool:
     return False
   return True
 
+"""
+Get the effective STP protocol for a node, following the hierarchy:
+node-level override → topology-level → default 'stp'
+"""
+def get_effective_stp_protocol(node: Box, topology: Box) -> str:
```

Comment:
We have "modules.get_effective_module_attribute" function to get this job done (assuming you need it)

Admittedly, you'd have to add "or 'stp'" at the end as it can return None.